src/fs_terminal_commands.c

```diff
@@ -0,0 +1,244 @@
+/*
+Copyright (C) 2015 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+
+#include "console.h"
+#include "string.h"
+#include "iso.h"
+#include "fs_terminal_commands.h"
+
+void cat_file(const char *fname);
+void get_abs_path(const char *path, char *abs_path);
+void ls_dir(const char *dname);
+void move_into_directory(const char *dname, char *abs_path);
+void move_up_directory(char *abs_path);
+void process_path_part(const char *dname, char *abs_path);
+
+/**
+ * @brief Prints contents of a single file
+ * @param fname Name of file to be printed
+ */
+void cat_file(const char *fname) {
+    char abs_path[256];
+    char c;
+    get_abs_path(fname, abs_path);
+    struct iso_file *file = iso_fopen(abs_path, 3);
+    if (!file) {
+        console_printf("cat: %s does not exist\n", fname);
+        iso_fclose(file);
+        return;
+    }
+    while (!file->at_EOF) {
+        iso_fread(&c, 1, 1, file);
+        console_putchar(c);
+    }
+    iso_fclose(file);
+}
+
+/**
+ * @brief Translate a path argument into an absolute path
+ * @details Takes filepath path and translates it into an
+ * absolute path that is copied into abs_path. If the path
+ * is relative, it builds off of the cur_path. Otherwise, it
+ * treats path as an absolute path.
+ * @param path Argument that points to the desired file/directory
+ * @param abs_path Buffer into which resultant absolute path is placed
+ */
+void get_abs_path(const char *path, char *abs_path) {
+    if (strlen(path) == 0) {
+        abs_path[0] = '\0';
+        return;
+    }
+    if (strcmp(path, ROOT) == 0) {
+        strcpy(abs_path, ROOT);
+        return;
+    }
+    char path_copy[256];
+    strcpy(path_copy, path);
+    char *next_dir;
+
+    if (path[0] == '/') {   // Absolute path
+        strcpy(abs_path, "/");
+        next_dir = strtok(path_copy + 1, "/");    // Move up one character to look like a relative path in root directory
+    } else {    // Relative path
+        strcpy(abs_path, cur_path);
```

Comment:
right but say `cur_path` is 256 and you have a relative path that is not empty. that will overwrite past the size of abs_path (which is generally 256, but not necessarily). it would be helpful to add a size param.